mmv1/third_party/terraform/utils/common_diff_suppress.go.erb

```diff
@@ -164,4 +164,33 @@ func timestampDiffSuppress(format string) schema.SchemaDiffSuppressFunc {
 // this happens for an internal ip for Private Services Connect
 func internalIpDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
 	return (net.ParseIP(old) != nil) && (net.ParseIP(new) == nil)
-}
\ No newline at end of file
+}
+
+// suppress the diff between { } and {[]}
+func emptyOrEmptySetDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+	o, n := d.GetChange(strings.TrimSuffix(k, ".#"))
+	var l []interface{}
+	if old == "0" && new == "1" {
+		l = n.([]interface{})
+	} else if new == "0" && old == "1" {
+		l = o.([]interface{})
+	} else {
+		// we don't have one set and one unset, so don't suppress the diff
+		return false
+	}
+
+	if l[0] != nil {
+		contents := l[0].(map[string]interface{})
+		for _, v := range contents {
+			if reflect.TypeOf(v).String() == "*schema.Set" {
```

Comment:
I don't think we want to ever compare based on the string representation of the type.

Do we expect there to be anything other than a Set in here, or can we just assume it is either a Set or empty?

Reply:
@slevenick yes. Without it, it fails in making below changes

```
  service_account {
  }
```
to 
```
  service_account {
    scopes = []
  }
```

Reply:
I mean we should be able to check the type without comparing the string representation of it.

What are the possible values of this map, is it possible for this to be any type other than a set?

Reply:
@slevenick 
Below is complete schema. email is optional while scopes is required. Everything seems fine except the case for above update 
```
  service_account {
    email  = "test-service-account@test-project.iam.gserviceaccount.com"
    scopes = []
  }
```

Reply:
I think we would prefer a switch statement on v.Type()

I don't like converting the type to a string and comparing based on the string value. We should do something similar to: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/third_party/terraform/utils/datasource_helpers.go#L24

Reply:
Interface does not support Kind(). Changed the suppress function to specific `scopes`